src/core/gui/dialog/PageTemplateDialog.cpp

```diff
@@ -13,40 +13,63 @@
 
-PageTemplateDialog::PageTemplateDialog(GladeSearchpath* gladeSearchPath, Settings* settings, PageTypeHandler* types):
-        GladeGui(gladeSearchPath, "pageTemplate.glade", "templateDialog"),
-        settings(settings),
-        types(types),
-        pageTypeSelectionMenu(std::make_unique<PageTypeSelectionPopoverGridOnly>(types, settings, this)),
-        popupMenuButton(new PopupMenuButton(get("btBackgroundDropdown"), pageTypeSelectionMenu->getPopover())) {
+constexpr auto UI_FILE = "pageTemplate.glade";
+constexpr auto UI_DIALOG_NAME = "templateDialog";
+
+using namespace xoj::popup;
+
+PageTemplateDialog::PageTemplateDialog(GladeSearchpath* gladeSearchPath, Settings* settings, ToolMenuHandler* toolmenu,
+                                       PageTypeHandler* types):
+        gladeSearchPath(gladeSearchPath), settings(settings), toolMenuHandler(toolmenu), types(types) {
     model.parse(settings->getPageTemplate());
 
-    g_signal_connect(
-            get("btChangePaperSize"), "clicked",
-            G_CALLBACK(+[](GtkToggleButton* togglebutton, PageTemplateDialog* self) { self->showPageSizeDialog(); }),
-            this);
+    Builder builder(gladeSearchPath, UI_FILE);
+    window.reset(GTK_WINDOW(builder.get(UI_DIALOG_NAME)));
+
+    // Needs to be initialized after this->window
+    pageTypeSelectionMenu = std::make_unique<PageTypeSelectionPopoverGridOnly>(types, settings, this);
+    gtk_menu_button_set_popup(GTK_MENU_BUTTON(builder.get("btBackgroundDropdown")),
+                              pageTypeSelectionMenu->getPopover());
+
+    pageSizeLabel = GTK_LABEL(builder.get("lbPageSize"));
+    backgroundTypeLabel = GTK_LABEL(builder.get("lbBackgroundType"));
+    backgroundColorChooser = GTK_COLOR_CHOOSER(builder.get("cbBackgroundButton"));
+    copyLastPageButton = GTK_TOGGLE_BUTTON(builder.get("cbCopyLastPage"));
+    copyLastPageSizeButton = GTK_TOGGLE_BUTTON(builder.get("cbCopyLastPageSize"));
+
+
+    g_signal_connect_swapped(builder.get("btChangePaperSize"), "clicked",
+                             G_CALLBACK(+[](PageTemplateDialog* self) { self->showPageSizeDialog(); }), this);
+
+    g_signal_connect_swapped(builder.get("btLoad"), "clicked",
+                             G_CALLBACK(+[](PageTemplateDialog* self) { self->loadFromFile(); }), this);
 
-    g_signal_connect(get("btLoad"), "clicked",
-                     G_CALLBACK(+[](GtkToggleButton* togglebutton, PageTemplateDialog* self) { self->loadFromFile(); }),
-                     this);
+    g_signal_connect_swapped(builder.get("btSave"), "clicked",
+                             G_CALLBACK(+[](PageTemplateDialog* self) { self->saveToFile(); }), this);
 
-    g_signal_connect(get("btSave"), "clicked",
-                     G_CALLBACK(+[](GtkToggleButton* togglebutton, PageTemplateDialog* self) { self->saveToFile(); }),
-                     this);
+    g_signal_connect_swapped(builder.get("btCancel"), "clicked", G_CALLBACK(gtk_window_close), this->window.get());
```

Comment:
It looks more consistent to use the `getWindow()` method.
```suggestion
    g_signal_connect_swapped(builder.get("btCancel"), "clicked", G_CALLBACK(gtk_window_close), this->getWindow());
```